### DISCUSSION TITLE
Compile away overrides idea

### DISCUSSION THREAD

**Discussion by roblevintennis** [argyleink/open-props#119] [Q&A] [answered]
This is an "idea" but Q&A is easier to close out 😛 

I think for some of us anal folks that mostly want to define our own values e.g. we work with a creative director that would never allow us to use prebaked values for the brand, etc., it would be nice to have a sort of free for all `props.overrides.js` file that you leave blank with comment instructions where we can basically override any OP defined values as it gets processed last in the pipeline.

99% sure I can hack this in myself on a forked version, but I think having it as a blessed approach might be interesting and add to the value prop of OP.

**Comment by argyleink** [comment]
override like this?

```css
:root {
  --size-1: 10px;
  --size-2: 20px;
}
```

OP ships with 0 specificity so it's easy to override and wont be something to wrestle with. 

But you're hoping to do that ahead of time, like as a prebuilt file with your mods? 
could fork the repo too? npm install from your repo?

**Reply by roblevintennis** [reply]
Regarding overrides, if I understand you end up with two `--size-1` and the override does in fact "win", but you have two of them in the CSS. If you do that ahead of time at build you only have one which is preferred.

> could fork the repo too?

Yes, as I mentioned but not ideal.

Probably this just won't be a thing the way I'm suggesting since most likely case is you npm install and just use the properties.

**Comment by roblevintennis** [comment, marked as answer]
I think it makes most sense for someone with same motives to customize certain values to just cherry pick the OP modules you feel are helpful as is via imports. Then, just go with your own custom colors, spacing, or what not that you don't want prescribed values for.

**Reply by argyleink** [reply]
toooootally. i imagine a lot of folks can reach in for the easings and shadows, for example. or just the animations. cool, glad you're finding your needs are met.
